File: Convert BST to Greater Tree/ConvertBSTToGreaterTree.cs
namespace LeetcodePracticeCsharpVersion
{
    class ConvertBSTToGreaterTree
    {
        int currentTotal = 0;
        public TreeNode ConvertBST(TreeNode root)
        {
            PostPorder(root);

            return root;
        }

        private void PostPorder(TreeNode node)
        {
            if (node == null) return;
            PostPorder(node.right);

            node.val += currentTotal;
            currentTotal = node.val;

            PostPorder(node.left);
        }
    }
}

File: 0538_Convert BST to Greater Tree/ConvertBSTToGreaterTree.java
class Solution {
    private int accumulatedVal = 0;
    public TreeNode convertBST(TreeNode root) {
        PostOrder(root);
        return root;
    }
    
    private void PostOrder(TreeNode node){
        if(node == null) return;
        PostOrder(node.right);
        node.val += accumulatedVal;
        accumulatedVal = node.val;
        PostOrder(node.left);
    }
}